runtime/tests/sharedclassapi/sharedClasses.c

```diff
@@ -0,0 +1,538 @@
+ static char* msg30 = "Parsing option string '%s'";
+ static char* msg31 = "The default cache directory will be used";
+ static char* msg32 = "cache Directory to be searched is '%s'";
+ static char* msg33 = "Invalid value specified for expectedCacheCount. Expected an integer, got '%s'";
+ static char* msg34 = "Number of caches found will not be checked";
+ static char* msg35 = "Number of caches found will be checked. Expected value is %d";
+ static char* msg36 = "Command line values for the shared classes cache directory will be used";
+ static char* msg37 = "Command line values for the shared classes cache directory will be ignored";
+ static char* msg38 = "Will attempt to delete the caches after querying them";
+ static char* msg39 = "Caches will not be deleted after querying them";
+ static char* msg40 = "Cache prefix was specified as: '%s'. Only caches with this prefix will be deleted";
+ static char* msg41 = "No Cache prefix was specified. All caches will be deleted";
+ static char* msg42 = "onload processing complete\n\n";
+ static char* msg43 = "Unexpected return code '%d' from '%s'\n\n";
+ #if defined(ZOS)
+ #pragma convlit(suspend)
+ #endif /* defined(ZOS) */
+ /**
+  * Returns a newly allocated copy of original. If length is
+  * -1, then, copies the whole of original. If length is > 0
+  * and less then length of original, then only that number
+  * of chars will be copied. The length of the returned string
+  * will be 'length' + the null terminator
+  *
+  * Returns NULL if malloc fails.
+  *
+  */
+  char *
+  copyString(char *original, int length)
+  {
+     char *newCopy = NULL;
+     if (-1 == length) {
+         newCopy = malloc(strlen(original) + 1);
+         if (NULL == newCopy) {
+             return NULL;
+        }
+        strcpy(newCopy, original);
+        return newCopy;
+    }
+     newCopy = malloc(length + 1);
+     if (newCopy == NULL) {
+         return newCopy;
+     }
+     strncpy(newCopy, original, length);
+     newCopy[length] = '\0';
+     return newCopy;
+ }
+ /**
+  * This should be called as a pair with freeConvertedString
+  *
+  * Takes an ascii string.
+  * On Z/OS, copies the string, converts to ebcdic, and returns
+  * the new, copied string.
+  *
+  * On other platforms, returns the passed in string.
+  *
+  */
+ char* copyAndConvert(char* asciiString) {
+     #if defined(ZOS)
+         char *copiedString = copyString(asciiString, -1);
+         if (NULL != copiedString) {
+             __atoe(copiedString);
+         }
+         return copiedString;
+     #else
+         return asciiString;
+     #endif /* defined(ZOS) */
+ }
+ /**
+  * This should be called after copyAndConvert to free the copied
+  * string.
+  *
+  * On Z/OS will free the passed string
+  *
+  * On other platforms, this is a no-op.
+  */
+ void freeConvertedString(char* ebcdicString) {
+     #if defined(ZOS)
+            free(ebcdicString);
+    #endif /* defined(ZOS) */
+    return;
+ }
```

Comment:
Please don't indent preprocessor directives; omit the explicit return from functions returning void.
```c
void
freeConvertedString(char *ebcdicString)
{
#if defined(ZOS)
	free(ebcdicString);
#endif /* defined(ZOS) */
}
```